tempodb/wal/append_block.go

```diff
@@ -238,3 +243,20 @@ func (a *AppendBlock) file() (*os.File, error) {
 
 	return a.readFile, err
 }
+
+func (a *AppendBlock) adjustTimeRangeForSlack(start uint32, end uint32, additionalStartSlack time.Duration) (uint32, uint32) {
+	now := time.Now()
+	startOfRange := uint32(now.Add(-a.ingestionSlack).Add(-additionalStartSlack).Unix())
+	endOfRange := uint32(now.Add(a.ingestionSlack).Unix())
+
+	if start < startOfRange {
+		metricWarnings.WithLabelValues(a.meta.TenantID, reasonOutsideIngestionSlack).Inc()
+		start = uint32(now.Unix())
+	}
+	if end > endOfRange {
+		metricWarnings.WithLabelValues(a.meta.TenantID, reasonOutsideIngestionSlack).Inc()
```

Comment:
Not necessarily a problem, but noting that a single trace can trigger both start and end warnings.  Probably worth rearranging this logic to fix.